src/modules/edit.js

```diff
@@ -0,0 +1,28 @@
+import saveTodos from './save.js';
+import renderTodoList from './displaytask.js';
+
+const editTodoDescription = (task, todoList) => {
+  const inputEl = document.createElement('input');
+  inputEl.type = 'text';
+  inputEl.value = task.description;
+  inputEl.classList.add('edit-input');
+
+  const listItemEl = todoList.childNodes[task.index];
+  listItemEl.replaceChild(inputEl, listItemEl.children[1]);
+  inputEl.select();
+
+  inputEl.addEventListener('keydown', (event) => {
+    const local = JSON.parse(localStorage.getItem('tasks'));
+    const id = task.index;
+    if (event.key === 'Enter') {
+      task.description = inputEl.value.trim();
+      local[id].description = task.description;
+      saveTodos(local);
+      renderTodoList(todoList, local);
+    } else if (event.key === 'Escape') {
+      renderTodoList(todoList);
+    }
+  });
+};
+
+export default editTodoDescription;
```

Comment:
- [ ] [ OPTIONAL ] When I edit a task it actually saves a new one the older one stay also that bad for the user experience it would be better to edit the task directly without creating a new one